src/CheckedFile.cpp

```diff
@@ -56,6 +56,12 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#elif defined( __EMSCRIPTEN__ )
```

Comment:
Could you please include this with with the `__linux__` one above since it's the same?

This whole #ifdef nightmare needs to be cleaned up at some point...